tests/at_end2end_test/test/enrollment_setup.dart

```diff
@@ -0,0 +1,239 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:at_auth/at_auth.dart';
+import 'package:at_chops/at_chops.dart';
+import 'package:at_client/at_client.dart';
+import 'package:at_end2end_test/config/config_util.dart';
+import 'package:at_end2end_test/src/test_initializers.dart';
+import 'package:at_lookup/at_lookup.dart';
+import 'package:test/test.dart';
+import 'package:uuid/uuid.dart';
+
+/// Intentionally did not prefix/suffix the file name with test to refrain from running this in the test suite.
+
+const String pkamPublicKey = 'aesPkamPublicKey';
+const String pkamPrivateKey = 'aesPkamPrivateKey';
+const String encryptionPublicKey = 'aesEncryptPublicKey';
+const String encryptionPrivateKey = 'aesEncryptPrivateKey';
+const String selfEncryptionKey = 'selfEncryptionKey';
+const String apkamSymmetricKey = 'apkamSymmetricKey';
+const String enrollmentId = 'enrollmentId';
+
+void main() {
+  List atSignList = ConfigUtil.getYaml()['enrollment']['atsignList'];
+  String namespace = 'wavi';
+
+  // Complete the initial sync for submitting the enrollment request to prevent time-out issues.
+  setUpAll(() async {
+    for (var atSign in atSignList) {
+      await TestSuiteInitializer.getInstance()
+          .testInitializer(atSign, namespace, authType: 'pkam');
+    }
+  });
+
+  for (var currentAtSign in atSignList) {
+    test('A test to submit and approve an enrollment for $currentAtSign',
+        () async {
+      // Set SPP at the start of enrollment tests to pass as OTP.
+      await TestSuiteInitializer.getInstance()
+          .testInitializer(currentAtSign, namespace, authType: 'pkam');
+      // Set SPP into the Remote Secondary
+      var atResponse =
+          await AtClientManager.getInstance().atClient.setSPP('ABC123');
+      expect(atResponse.response, 'ok');
+
+      // Submit an enrollment request with at_auth package
+      AtEnrollmentBase atEnrollmentBase =
+          atAuthBase.atEnrollment(currentAtSign);
+      int random = Uuid().v4().hashCode;
+      AtLookUp atLookUp = AtLookupImpl(
+          currentAtSign,
+          AtClientManager.getInstance().atClient.getPreferences()!.rootDomain,
+          AtClientManager.getInstance().atClient.getPreferences()!.rootPort);
+
+      EnrollmentRequest enrollmentRequest = EnrollmentRequest(
+          appName: 'wavi-$random',
+          deviceName: 'iphone',
+          otp: 'ABC123',
+          namespaces: {'wavi': 'rw', 'buzz': 'rw'});
```

Comment:
We should clean up enrollments from previous test runs. In apkam_setup.sh the NoPorts e2e tests we do the following before submitting the new enrollment request for this test run:
```
  logInfo "Denying any pending enrollment requests for $atSign with apkamAppName $apkamApp and apkamDeviceName $apkamDev"
  $authBinary deny -r "$atDirectoryHost" -a "$atSign" --arx "$apkamApp" --drx "$apkamDev" || return $?

  logInfo "Revoking any approved enrollment requests for $atSign with apkamAppName $apkamApp and apkamDeviceName $apkamDev"
  $authBinary revoke -r "$atDirectoryHost" -a "$atSign" --arx "$apkamApp" --drx "$apkamDev" || return $?
```
where `$apkamApp` has been set to "e2e_all"

For these tests I suggest
- use an apkam appName of 'e2e_test'
- before submitting the new enrollment request:
  - fetch all pending enrollment requests where the appName is 'e2e_test', and deny all of them
  - fetch all approved enrollment requests where the appName is 'e2e_test', and revoke all of them